image-roll.el

```diff
@@ -467,30 +489,44 @@ When SCREEN is non-nil, scroll by window height."
 
 
     (if (cond ((< new-page current-page)
-               (goto-char (image-roll-page-to-pos new-page))
+               (when (>= new-page 1)
+		 (run-hooks 'image-roll-before-change-page-hook))
                (setf (image-roll-current-page) (max new-page 1))
-               (image-set-window-vscroll
-                (if (< new-page 1)
-                    (user-error "Beginning of document")
-
-                  (- (image-roll-overlay-height (image-roll-current-page))
-                     remaining-height))))
+               (when (>= new-page 1)
+		 (run-hooks 'image-roll-change-page-hook))
+               (goto-char (image-roll-page-to-pos new-page))
+               (prog1
```

Comment:
Is there a particular reason why you are using `progn1` here? I think the return value is not used, so it is somewhat confusing to 'suggest' that.